.github/workflows/release.yml

```diff
@@ -0,0 +1,158 @@
+name: Build Release
+# Run on semver tags.
+on:
+  push:
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+permissions:
+  contents: read
+  packages: write # Allow docker/build-push-action to publish to GitHub Container Registry
+  pull-requests: read
```

Comment:
I don't think this one needs access to PRs, since it's only running on tag pushes